### PR TITLE
Rafactor.

### PR DESCRIPTION
**Pull request by cmkohnen**
No longer call toolbars inbuilt functions to store view states. Rely on own mode implementation inside toolbar logic.

fix linting.

tox.ini: allow W503

Use @property decorator for setting and getting canvas limits

### CODE REVIEW

File: src/canvas.py [hunk 1]
```diff
@@ -273,6 +276,37 @@ def set_legend(self):
...
+                self.canvas.set_cursor(tools.Cursors.SELECT_REGION)
+                self._last_cursor = tools.Cursors.SELECT_REGION
+            elif (mode == InteractionMode.PAN
+                  and self._last_cursor != tools.Cursors.MOVE):
+                self.canvas.set_cursor(tools.Cursors.MOVE)
+                self._last_cursor = tools.Cursors.MOVE
+        elif self._last_cursor != tools.Cursors.POINTER:
+            self.canvas.set_cursor(tools.Cursors.POINTER)
+            self._last_cursor = tools.Cursors.POINTER
+
```
Comment (Sjoerd1993) [inline src/canvas.py:309]: What do these two overwritten function change compared to their original counterparts?

File: src/canvas.py [hunk 2]
```diff
@@ -287,20 +321,9 @@ def remove_rubberband(self):
 
     # Overwritten function - do not change name
     def push_current(self):
-        super().push_current()
         self.canvas.apply_limits()
```
Comment (Sjoerd1993) [inline src/canvas.py:324]: Yeah running the base class version is just a waste of resources at this point

File: tox.ini [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [flake8]
 count = True
 #default_ignorelist: E121,E123,E126,E226,E24,E704,W503,W504
-ignore = U101, SCS109, SCS101, SCS113
+ignore = U101, SCS109, SCS101, SCS113, W503
```
Comment (Sjoerd1993) [inline tox.ini:4]: Yeah we should either disable W503 or W504, as they are kinda in direct conflict with each other (they describe each others best practice as antipattern and vice versa)

https://www.flake8rules.com/rules/W503.html
https://www.flake8rules.com/rules/W504.html

Not sure if I have a very strong preference tbh

**Review by Sjoerd1993 — APPROVED** [review]

File: src/canvas.py (hunk 1 above)
Comment (cmkohnen) [reply src/canvas.py:309]: The base implementation checks against the mode stored inside the toolbar itself. The difference here is, that we check against InteractionMode stored inside the main application, deduplicating logic

**Review by Sjoerd1993 — APPROVED** [review]

**Pull request merged**